src/main/java/ch/njol/skript/util/Utils.java

```diff
@@ -285,14 +293,55 @@ public static File getFile(Plugin plugin) {
 	public static NonNullPair<String, Boolean> getEnglishPlural(String word) {
 		assert word != null;
 		if (word.isEmpty())
-			return new NonNullPair<>("", Boolean.FALSE);
+			return new NonNullPair<>("", false);
+		if (!couldBeSingular(word)) {
+			for (final WordEnding ending : plurals) {
+				if (ending.isCompleteWord()) {
+					// Complete words shouldn't be used as partial pieces
+					if (word.length() != ending.plural().length())
+						continue;
+				}
+				if (word.endsWith(ending.plural()))
+					return new NonNullPair<>(
+						word.substring(0, word.length() - ending.plural().length()) + ending.singular(),
+						true
+					);
+				if (word.endsWith(ending.plural().toUpperCase(Locale.ENGLISH)))
+					return new NonNullPair<>(
+						word.substring(0, word.length() - ending.plural().length())
+							+ ending.singular().toUpperCase(Locale.ENGLISH),
+						true
+					);
+			}
+		}
+		return new NonNullPair<>(word, false);
+	}
+
+	private static boolean couldBeSingular(String word) {//todo
 		for (final WordEnding ending : plurals) {
-			if (word.endsWith(ending.plural()))
-				return new NonNullPair<>(word.substring(0, word.length() - ending.plural().length()) + ending.singular(), Boolean.TRUE);
-			if (word.endsWith(ending.plural().toUpperCase(Locale.ENGLISH)))
-				return new NonNullPair<>(word.substring(0, word.length() - ending.plural().length()) + ending.singular().toUpperCase(Locale.ENGLISH), Boolean.TRUE);
+			if (ending.singular().isBlank())
+				continue;
+			if (ending.isCompleteWord() && ending.singular().length() != word.length())
+				continue; // Skip complete words
+
+			if (word.endsWith(ending.singular()) || word.toLowerCase().endsWith(ending.singular())) {
+				return true;
+			}
 		}
```

Comment:
Git chose an insane way to diff this. It's two separate methods, in case that isn't clear.